### PR TITLE
PP-839 Serialise service name and card types in charge object

### PR DESCRIPTION
**Pull request by simad**
## WHAT
- `service_name` and `card_type` were already embedded in the charge object, but they were not  serialised in any frontend endpoint. 
- We have now two different `JsonViews` which allow us to serialise GatewayAccountEntity in a different way, depending from the resource. This is needed because both of those values have to appear in FrontEnd and not in PublicApi.
## HOW

_Steps to test or reproduce:_

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesFrontendResourceITest.java [hunk 1]
```diff
@@ -362,35 +364,51 @@ private ValidatableResponse validateGetCharge(long expectedAmount, String charge
...
-                .body("confirmation_details.expiry_date", is("03/18"))
-                .body("confirmation_details.billing_address", is(notNullValue()))
-                .body("confirmation_details.billing_address.line1", is("line1"))
-                .body("confirmation_details.billing_address.line2", is(nullValue()))
-                .body("confirmation_details.billing_address.city", is("city"))
-                .body("confirmation_details.billing_address.postcode", is("postcode"))
-                .body("confirmation_details.billing_address.country", is("country"))
-                .body("confirmation_details.billing_address.county", is(nullValue()));
 
+    private void validateGatewayAccount(ValidatableResponse response) {
```
Comment (rauligs) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesFrontendResourceITest.java]: Can be asserted the other fields are not being serialised?

File: src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceITest.java [hunk 2]
```diff
@@ -121,6 +125,20 @@ public void shouldGetGatewayAccountForExistingAccount() {
...
+        String accountId = createAGatewayAccountFor("worldpay");
+        String frontendCardTypeUrl = ACCOUNTS_CARD_TYPE_FRONTEND_URL.replace("{accountId}", accountId);
+        GatewayAccountPayload gatewayAccountPayload = GatewayAccountPayload.createDefault().withMerchantId("a-merchant-id");
+        app.getDatabaseTestHelper().updateCredentialsFor(Long.valueOf(accountId), gson.toJson(gatewayAccountPayload.getCredentials()));
+        app.getDatabaseTestHelper().updateServiceNameFor(Long.valueOf(accountId), gatewayAccountPayload.getServiceName());
+        givenSetup().accept(JSON)
+                .get(frontendCardTypeUrl)
+                .then()
+                .statusCode(200)
+                .body("card_types", is(notNullValue()));
```
Comment (rauligs) [inline src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceITest.java]: content assertions? unless are done for the same resource somewhere else.

Comment (simad) [reply src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceITest.java]: 👍 adding them!

**Comment by simad** [comment]
Addressed comments, see https://github.com/alphagov/pay-connector/pull/238/commits/7f562a492c52b9acf1c89f91439e8b4438595400

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesFrontendResourceITest.java (hunk 1 above)
Comment (rauligs) [reply src/test/java/uk/gov/pay/connector/it/resources/ChargesFrontendResourceITest.java]: The correct assertion for containsKey is:

`
.body("containsKey('gateway_account')", is(true))
.body("gateway_account.containsKey('gateway_account_id')", is(false))
.body("gateway_account.containsKey('type')", is(false))
.body("gateway_account.containsKey('payment_provider')", is(false))
.body("gateway_account.containsKey('credentials')", is(false))
`

If not these are false positives

Comment (simad) [reply src/test/java/uk/gov/pay/connector/it/resources/ChargesFrontendResourceITest.java]: Well spotted 👍

**Comment by PauloPortugal** [comment]
rebuild

**Comment by PauloPortugal** [comment]
rebuild

**Pull request merged**